pkg/urbit/daemon/main.c

```diff
@@ -1196,10 +1196,32 @@ _cw_utils(c3_i argc, c3_c* argv[])
   return 0;
 }
 
+/* _main_bin_path(): copy binary path, ensuring ".exe" ending on windows
+*/
+static c3_c*
+_main_bin_path(c3_c* bin_c)
+{
+#ifdef U3_OS_mingw
+  c3_w len_w = strlen(bin_c);
+
+  if ( (4 <= len_w) && !strcmp(bin_c + len_w - 4, ".exe") ) {
+    c3_c* pat_c = c3_malloc(5 + len_w);
+    sprintf(pat_c, "%s.exe", bin_c);
+    return pat_c;
+  }
+#endif
+
+  return strdup(bin_c);
+}
```

Comment:
This doesn't seem to work if `strlen(bin_c) < 4`. I know it's unlikely that we have binary names shorter than four characters, but I think we should still get this right for all cases. Based off preliminary testing, the following should work for all possible lengths of `bin_c`.
```c
#ifdef U3_OS_mingw
  static const c3_c exe_c[] = ".exe";
  size_t len_i = strlen(bin_c);

  c3_c* dot_c = strrchr(bin_c, '.');
  if ( !dot_c || 0 != strcmp(dot_c, exe_c) ) {
    len_i += sizeof(exe_c);
    c3_c* pat_c = c3_malloc(len_i);
    snprintf(pat_c, len_i, "%s%s", bin_c, exe_c);
    return pat_c;
  }
#endif
  return strdup(bin_c);
}
```